### ISSUE TITLE
'answerCall' Listener occasionally does not run, when answering call while app is terminated/background

### ISSUE THREAD

**Issue by stephanoparaskeva** [react-native-webrtc/react-native-callkeep#339]
'runs pickup' is never logged to console. 
I suspect the 'answerCall' event occurs before listener is added in some cases. How do you check if app loaded with 'answerCall' event. As opposed to detecting an event in real time?

IE If 'answerCall' has already occured, then after you add the event listener: RNCallKeep.addEventListener('answerCall', pickupCall), the function will instantly run. Or a way to check if 'answerCall' has already occured?

```ts
  // Picks up and accepts incoming call.
  const pickupCall = async () => {
    console.warn('runs pickup'); // <--------- Never Runs
    setOpenCall({ recipient: {}, inCall: true, isIncomingCall: true });
    const id = oneTimeId();
    // Tell callkeep that call has begun
    RNCallKeep.setCurrentCallActive(id);

    // Relay that phone has been picked up
    // Just incase this function runs before 'onCallListener'
    // If it runs before 'sess.current' will not be the real session
    // So methods below will not do anything.
    pickedUp.current = true;

    // onRemoteStreamListener event gets registered if both phones run this function
    const stream = await sess.current?.getUserMedia?.(MEDIA_OPTIONS);
    // The function to cause the call to be connected
    await sess.current?.accept?.({});
    // Setting of localStream video and preparing the remote stream object for incoming remote stream
    setLocalStreams([{ userId: 'localStream', stream }]);
    setRemoteStreams([{ userId: sess.current.initiatorID, stream: null }]);
  };

```

```ts
  const callkeepListenersInit = async () => {
    console.warn('runs callkeep');
    RNCallKeep.setAvailable(true);
    RNCallKeep.addEventListener('answerCall', pickupCall);
    RNCallKeep.addEventListener('endCall', rejectCall);
  };

  // Removes all listeners when component unmounts.
  const cleanup = () => {
    RNCallKeep.removeEventListener('answerCall');
    RNCallKeep.removeEventListener('endCall');
    hangup();
  };

  // First mount hooks up all listeners.
  useEffect(() => {
    cubeListenersInit();
    callkeepListenersInit();
    return cleanup;
  }, []);
```

**Comment by namnm** [comment]
May be related to this: https://github.com/react-native-webrtc/react-native-callkeep#--didloadwithevents

**Comment by stephanoparaskeva** [comment]
> May be related to this: https://github.com/react-native-webrtc/react-native-callkeep#--didloadwithevents

This is exactly what I was looking for, great.

There's the residual problem that when using didLoadWithEvents, the call never gets put in 'connected' state and so the user cannot hear anything. Just the dial tone. How does one move the call into connected state when using didLoadWithEvents

**Comment by namnm** [comment]
I dont know much about your detail situation so I can not help.

About me, I handled that didLoadWithEvents as following: https://github.com/brekekesoftware/brekekephone/blob/master/src/utils/callkeep.ts#L62

**Comment by agamsps** [comment]
Hi guys, @namnm  @stephanoparaskeva 
I have a problem i am following from quite a few day.
My app ( on answering the call from call kit UI on iPhone ) on click, is opening the app when in debug mode but does not open app in release mode .
By debug mode i mean to say when debugger ison ( edit sceme in xcode is debug )

I will explain my scenario in more depth,

I have a screen getting called in the event of 
```
RNCallKeep.addEventListener('answerCall', ({ callUUID }) => {
// i am navigating the user to a particular screen
    });
```
In debug mode :  the app opens when i answer the call, it navigates to the screen to which i have navigated to the screen in the above event listener.
In release mode :  the app **does not** open when i answer the call, But when i manually open the app, it is already navigated to that screen, means it was already fired.

So the main issue is that the answer button is not opening the app when it is in background/ killed.

For more info : i am using voip push notification and triggering to display incoming call via native code like below : 

`- (void)pushRegistry:(PKPushRegistry *)registry didReceiveIncomingPushWithPayload:(PKPushPayload *)payload forType:(PKPushType)type withCompletionHandler:(void (^)(void))completion {
  
  NSLog(@"printing notification: %@", [payload dictionaryPayload]);
  NSString *uuid = [[[payload dictionaryPayload] objectForKey:@"data"] objectForKey:@"channelUniqueName"];
  NSString *callerName = [[[payload dictionaryPayload] objectForKey:@"data"] objectForKey:@"callerName"];
  NSString *handle = [[[payload dictionaryPayload] objectForKey:@"data"] objectForKey:@"callerName"];
  NSString *notitype = [[[payload dictionaryPayload] objectForKey:@"data"] objectForKey:@"notificationType" ];
  
  // --- this is optional, only required if you want to call `completion()` on the js side
  [RNVoipPushNotificationManager addCompletionHandler:uuid completionHandler:completion];

  // --- Process the received push
  [RNVoipPushNotificationManager didReceiveIncomingPushWithPayload:payload forType:(NSString *)type];
  
    [RNCallKeep reportNewIncomingCall: uuid
                                 handle: handle
                             handleType: @"generic"
                               hasVideo: YES
                    localizedCallerName: callerName
                        supportsHolding: YES
                           supportsDTMF: YES
                       supportsGrouping: YES
                     supportsUngrouping: YES
                            fromPushKit: YES
                                payload: uuid
                  withCompletionHandler: completion];

  // --- You should make sure to report to callkit BEFORE execute `completion()`

  
  //if (@available(iOS 14, *)) {
    completion();
  //}
  // --- You don't need to call it if you stored `completion()` and will call it on the js side.
  
}`

Any kind of help would be appreciated .

**Comment by classiebit** [comment]
Someone, please help! 🙏 `answerCall` event is not firing in **Android** only. When we call `displayIncomingCall` and the incoming call shows on Android, and when we press the call end button, the `endCall` event is firing, but when we press the call-accept button, it stays there on the call screen and nothing happens. And the `answerCall` event is not firing.

Please help, we've tried all the solutions, nothing is working.

**Comment by prasanthvenkatachalam** [comment]
Have you fixed??. I am also facing the same issue

**Comment by classiebit** [comment]
No, it didn't work, and we just removed the rn-callkeep package from Android and we created a custom solution to receive calls on Android. 

We created a solution to receive calls via silent push notification and created a custom page to show incoming calls on Android.

**Comment by fukemy** [comment]
hi, i got this problem on IOS, not android
